### ISSUE TITLE
Cookbook import epub for reader view

### ISSUE THREAD

**Issue by pheutar** [TandoorRecipes/recipes#1147]
I have been using Tandoor as a replacement for `Nextcloud Recipes` and I LOVE IT! Thank you for all the hard work.

Forgive me if this is a ridiculous request but would it rule if we could start a discussion around adding a `ePub` or any digital book reader into Tandoor. As my skills with cooking grow, so do my cookbooks, and now I find myself flipping through page after page, index after index wondering "What book was that recipe in again". The idea of digitizing all these pages is a herculean task and I have done some of my all time faves but the mountain is growing. I recently have been buying my book collection over in `ePub` so I can actually search these things.

Would a project like [eplee](https://github.com/Janglee123/eplee) be a starting point? I'm not saying that the `ePub`'s need to be imported, scraped, and added into the recipes view like a webpage import. A simple tab that is just a clean rendered view of the book that has some basic search and maybe the ability to tag (in a perfect world) would be top notch!

**Comment by vabene1111** [comment]
thanks for the kind word and the interesting request. Tandoor actually started in the first place because i had hundreds of PDF which i wanted to tag and properly search, i think this is similar to your problem. My solution was first to just build a search and link to the dropbox where i stored the PDFs but today tandoor includes a web viewer for both PDF and images (see https://docs.tandoor.dev/features/external_recipes/).

do you think that this soultion could work for your ebooks as well ? is it possible to split individual recipes into single files ? how is DRM working here, do you have books that are unprotected and could thus be imported/rendered ? If so do you think you could send me a few test files to info@tandoor.dev i think i could try adding an "epub" reader similar to the PDF reader. 

alternatively i could image we could build some tools to convert an epub book into indivudual PDFs of recipes which could then be used with the external feature.

Interested to hear your feedback

**Comment by pheutar** [comment]
I will wrangle up some DRM free ePubs and send em' off your way and I will also spend some spare time this week looking at splitting the recipes to single files. I've not looked to closely at the ePub spec to see what's possible either but I do know that ePub to PDF is definitely possible. I'm not sure if it's possible with DRM protected books (which is a lot of what I own from purchasing from book publishers) but I think it is well worth looking into for myself.

**Comment by vabene1111** [comment]
awesome, feel free to sum up here what you find so that we can think together about how to best approach this.

**Comment by pheutar** [comment]
The easiest solution I have found since my books are all DRM protected was just to convert them to PDF and upload them as local media. I am having some issues making this work though. 

What I've done:
docker-compose -> added a path of -> $media_dir/pdf:/opt/recipes/pdf 
Inside tandoor I have added external storage -> named it, method: local, and gave it a full path of "/opt/recipes/pdf" 

Now when I add the external resource and "sync" it errors out... Am I doing something wrong?
